### ISSUE TITLE
Allow PrivateUse1 backends to opt out of CompositeExplicitAutograd decomposition

### ISSUE THREAD

**Issue by vishalgoyal316** [pytorch/pytorch#188052]
## Problem

PyTorch's dispatch table resolves **CompositeExplicitAutograd (CEA)** kernels before the backend fallback. PrivateUse1 backends that register a single catch-all fallback kernel never see composite ops — the dispatcher silently decomposes them into primitives before the fallback is invoked.

The dispatch resolution order for a PrivateUse1 backend is:

```
1. PrivateUse1 kernel          ← direct registration, always wins
2. CEANF alias kernel          ← silently decomposes (CompositeExplicitAutogradNonFunctional)
3. CEA alias kernel            ← silently decomposes (CompositeExplicitAutograd)  ← THE PROBLEM
4. CIA / Autograd alias        ← CompositeImplicitAutograd
5. Backend fallback            ← only reached if no alias kernel matched above
```

As a result, ops like the following are decomposed before the backend ever sees them:

| Composite op      | Decomposes to              | Alias key |
| ----------------- | -------------------------- | --------- |
| `convolution`     | `convolution_overrideable` | CEA       |
| `softmax.int_out` | `softmax_out`              | CEA       |

Steps 2 and 3 cannot currently be bypassed without registering a PrivateUse1-specific kernel for every affected op individually. Step 4 (CIA) is a separate mechanism and is intentionally not part of this proposal.

## Impact

**Who is affected:** PrivateUse1 backends that implement a single catch-all fallback kernel and forward ATen ops verbatim to an external execution environment — for example, a remote device, a hardware accelerator with its own op set, or a tracing backend. These backends need to receive composite ops whole; the external side often has optimized implementations for ops like `convolution` directly, not just the decomposed primitives like `convolution_overrideable`.

**Example pattern:**

```python
# A PrivateUse1 backend registers one fallback to handle all ops.
lib = torch.library.Library("_", "IMPL", "PrivateUse1")

def my_fallback(op, args, kwargs):
    # forward op to external execution environment (remote device, accelerator, etc.)
    external_device.run(op.name(), args, kwargs)

lib.fallback(my_fallback)
```

With this pattern, calling `torch.nn.functional.conv2d(tensor)` on a PrivateUse1 tensor results in `my_fallback` receiving `convolution_overrideable` — not `convolution`. If the external execution environment expects `convolution`, the backend breaks silently.

**Scale:**

Hundreds of ATen op overloads have CEA registrations. A proxy backend must manually register a stub for each one to prevent decomposition — a list that grows silently with every PyTorch release.

## Request

A backend-level opt-out mechanism so that PrivateUse1 backends can bypass CEA/CEANF decompositions and receive composite ops directly through the registered fallback. 

I am working on a fix and will raise a PR shortly.

cc: @mansiag05 

cc @NmomoN @mengpenghui @fwenguang @cdzhan @1274085042 @PHLens @albanD
